File: Sorting Algorithms/Bubble Sort.cpp

#include <iostream>
using namespace std;

void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; ++i) {                     // Outer Loop
        for (int j = 0; j < n - i - 1; ++j) {             // Inner Loop
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
            }
        }
    }
}

int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);
    bubbleSort(arr, n);
    
    cout << "Sorted array: ";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: bubble_sort.php
<?php

function bubbleSort($arr) {
	$n = count($arr);

	// Traverse through all array elements
	for ($i = 0; $i < $n - 1; $i++) {
		for ($j = 0; $j < $n - $i - 1; $j++) {

			// Swap if the element found is 
			// greater than the next element
			if ($arr[$j] > $arr[$j + 1]) {
				$temp = $arr[$j];
				$arr[$j] = $arr[$j + 1];
				$arr[$j + 1] = $temp;
			}
		}
	}

	return $arr;
}

// Driver code
$arr = [61, 33, 22, 14, 27, 15, 92];
$sortedArray = bubbleSort($arr);

// Print the sorted array
echo "Sorted array: "
	. implode(", ", $sortedArray);

?>